DoSomethingArchive/legacy-website
# #7392: Upgrade Metatag from 7.x-1.4 to 7.x-1.21

#### What's this PR do?

Upgrades the Metatag module from 7.x-1.4 to 7.x-1.21.

#### How should this be reviewed?

1. Pull the code
2. Run `ds build`
3. Compare metatags on your local to those on staging and verify that the only things that are different are the hostnames in them. Look at node detail pages especially when doing comparisons.
4. Edit a couple nodes just to confirm that the metatag options in forms still look fine. 

#### Any background context you want to provide?

Lots of update hooks ran here:

```
Metatag: 7036  Update variables to indicate which entities should be supported.
Metatag: 7037  Clear the menu cache so the renamed Settings page will be picked up.
Metatag: 7038  Manually enable all content types, vocabularies and the user entity to help  resolve issues from 1.5's architecture change.
Metatag: 7039  Reload some meta tag caches.
Metatag: 7040  Fix robots meta tags that might have been broken when they were imported from Nodewords.
Metatag: 7041  Rerun update 7018 that was fixed to run correctly when using Entity Translation. This may take some time.
Metatag: 7100  Delete a deprecated variable and clear all Metatag caches.
Metatag: 7101  Update i18n strings for all meta tags to use the new format.
Metatag: 7102  Re-run update 7101.
Metatag: 7103  Clear all metatag caches so the new entity caching structure takes over.
Metatag: 7104  Remove the entity revision ID from the translation strings.
Metatag: 7105  Fix the output translation strings.
Metatag: 7106  The output translation strings were renamed to something shorter, so rerun  update 7105.
Metatag: 7107  Fix the global config translation strings.
Metatag: 7108  Delete output translations if it's disabled.
Metatag: 7109  Rename the 'icon_any' meta tag to 'mask-icon' in the entity records.
Metatag: 7110  Rename the 'icon_any' meta tag to 'mask-icon' in the configs.
Metatag: 7111  Remove the "metatag_ui" record from the {system} table.
Metatag: 7112  The Publisher meta tag is now part of the Google Plus submodule.
Metatag: 7113  Clear all metatag output caches if output caching is disabled.
Metatag: 7114  Disable output caching.
Metatag: 7115  Clear all metatag output caches, it will be rebuild if needed.
Metatag_twitter_cards: 7100  Rename the 'twitter:image:src' meta tag back to 'twitter:image'.
Metatag_twitter_cards: 7101  Rename the 'twitter:image:src' meta tag back to 'twitter:image', part 2
Metatag_twitter_cards: 7102  Clear the Metatag cache.
```

They all ran successfully locally - no errors and no issues when testing afterwards. 

*Note that metatags maybe reordered when doing before/after tests, which is OK, as long as they're otherwise the same.*

I also skimmed the codebase for anything custom we're doing on top of metatag. The `dosomething_metatag` module has a decent amount of stuff but it mostly seems to be contained inside `hook_preprocess_page()` and still appears to work fine, after testing some of the metatags that are added there. The exported Features config for metatags in that module looks to be correct after the upgrade as well.

#### Relevant tickets
Fixes #7392
